x-pack/apm-server/sampling/eventstorage/storage.go

```diff
@@ -65,8 +67,9 @@ func (s *Storage) NewShardedReadWriter() *ShardedReadWriter {
 // The returned ReadWriter must be closed when it is no longer needed.
 func (s *Storage) NewReadWriter() *ReadWriter {
 	return &ReadWriter{
-		s:   s,
-		txn: s.db.NewTransaction(true),
+		s:           s,
+		txn:         s.db.NewTransaction(true),
+		pendingSize: defaultPendingSize,
```

Comment:
Instead of defaulting to 1024, should we perhaps use `s.db.Size()`? And rather than calculating `pendingSize`, maybe we could track the remaining capacity? i.e. `StorageLimitInBytes - Size()`